bundles/org.openhab.binding.nuki/README.md

```diff
@@ -34,35 +34,146 @@ The following configuration options are available:
+  | 0      | Uncalibared              |
+  | 1      | Locked                   |
+  | 2      | Unlocking                |
+  | 3      | Unlocked                 |
+  | 4      | Locking                  |
+  | 5      | Unlatched                |
+  | 6      | Unlatched (Lock 'n' Go)  |
+  | 7      | Unlatching               |
+  | 254    | Motor blocked            |
+  | 255    | Undefined                |
+
+  Unfortunately the Nuki Bridge is not reporting any transition states (e.g. for Lock 'n' Go).
 
 - **lowBattery** (Switch)  
     Use this channel to receive a low battery warning.
-
-- **doorsensorState** (Number)  
-    Use this channel if you want to display the current door state provided by the door sensor.  
-    Supported Door Sensor States are : `0` (Unavailable), `1` (Deactivated), `2` (Closed), `3` (Open), `4` (Unknown) and `5` (Calibrating).
+  
+- **keypadLowBattery** (Switch)
+    Use this channel to receive a low battery warning for keypad paired with smart lock.
+  
+- **batteryLevel** (Number)
+    Use this channel to monitor current battery level of smart lock.
+  
+- **batteryCharging** (Switch)
+    Use this channel to monitor charging of smart lock.
+
+- **doorsensorState** (Number)
+    Use this channel if you want to display the current door state provided by the door sensor.
+  
+  | Action | Name                     |
+  |--------|--------------------------|
+  | 0      | Unavailable              |
+  | 1      | Deactivated              |
+  | 2      | Closed                   |
+  | 3      | Open                     |
+  | 4      | Unknown                  |
+  | 5      | Calibrating              |
+
+### Nuki Opener
+
+Nuki Opener has no configuration properties.
+
+#### Supported channels
+
+- **openerState** (Number)
+  Use this channel if you want to execute supported Opener actions or monitor current Opener state.
+  Supported actions are:
+
+  | Action | Name                       |
+  |--------|----------------------------|
+  | 1      | Activate ring to open      |
+  | 2      | Deactivate ring to open    |
+  | 3      | Electric strike actuation  |
+  | 4      | Activate continuous mode   |
+  | 5      | Deactivate continuous mode |
+
+  Supported opener states are:
+
+  | State  | Name                |
+  |--------|---------------------|
+  | 0      | Untrained           |
+  | 1      | Online              |
+  | 3      | Ring to open active |
+  | 5      | Open                |
+  | 7      | Opening             |
+  | 253    | Boot run            |
+  | 255    | Undefined           |
+
+- **openerMode** (Number)
+  Use this channel to monitor opener modes. Supported values are:
+  
+  | Mode   | Name            |
+  |--------|-----------------|
+  | 2      | Door mode       |
+  | 3      | Continuous mode |
+
+- **openerLowBattery** (Switch)
+  Use this channel to receive a low battery warning.
+
+- **ringActionState** (Switch)
+  Use this channel to receive notification when doorbell rings.
```

Comment:
This seems rather a trigger channel. See https://www.openhab.org/docs/developer/bindings/thing-xml.html#trigger-channel-types

Reply:
I'm not sure. The bridge API works by sending http callback which contains state of all devices connected to given bridge (opener and smartlock) every time something changes and ringActionState is boolean property of opener. It seems to me that in order to implement it as a trigger channel, I'd have to somehow remember the last value of this flag for opener in order to properly trigger something like "ringing started" or "ringing stopped" action.

Reply:
Your description sounds as it is indeed a stateful channel. So, the switch is ON as long as the visitor pushes the doorbell. Then, your model is correct.

Reply:
Not exaclty, the official documentation says only "Flag indicating if a ring-action is currently occuring or not (reset after 30 seconds)". Looking through my openhab logs, it doesn't seem to work very well. Basically when someone rings the doorbell, nuki sets the flag to true and then makes HTTP call to webhook with complete state of all things connected to bridge (so you don't really know WHAT happened). Then after 30s its turned to false but no callback is made, so the only way the binding sets it back to false is if some other action which triggers callback happens.
Seems like I'll have to somehow incorporate that 30s timeout otherwise you won't get any new notifications after someone rang the doorbell for the first time unless some other action occured.

Reply:
That sounds like a poor API. If the 30s is a static value, you could make it a trigger channel, though.

Reply:
Switched to trigger channel that will trigger every time nuki sends status with ringActionState set to true. but only once every 30s.